src/memory_prefetch.c

```diff
@@ -382,17 +382,19 @@ int addCommandToBatch(client *c) {
 
     batch->clients[batch->client_count++] = c;
 
-    if (likely(c->iolookedcmd)) {
-        /* Get command's keys positions */
-        getKeysResult result = GETKEYS_RESULT_INIT;
-        int num_keys = getKeysFromCommand(c->iolookedcmd, c->argv, c->argc, &result);
-        for (int i = 0; i < num_keys && batch->key_count < batch->max_prefetch_size; i++) {
-            batch->keys[batch->key_count] = c->argv[result.keys[i].pos];
+    pendingCommand *pcmd = c->pending_cmds.head;
```

Comment:
here maybe we can check if `c->pending_cmds.len + batch->key_count` is more than batch->max_prefetch_size when batch->key_count > 0, return C_ERR if so, to avoid some keys not prefetching